From ab78a34158bdee7e8bffc600aff4f146ba30a14b Mon Sep 17 00:00:00 2001
From: Kenneth Graunke <kenneth@whitecape.org>
Date: Thu, 9 Mar 2017 11:47:43 -0800
Subject: [hack] i965: Use Y-tiling for window system buffers on Gen9+.

Y-tiling is much more efficient than X-tiling.  We've seen 10-20%
performance gains by doing this.

Unfortunately, we can't just enable this - the other components of
the graphics stack (kernel, DDX or wayland compositor) need to be able
to cope with it.  Daniel Stone is working on a real solution to this
negotiating problem.  But in the meantime, if you have a stack where
this works, then you can use this hack for better performance.

Either xf86-video-intel+SNA or modesetting+glamor with DRI3 should work.

Signed-off-by: Kenneth Graunke <kenneth@whitecape.org>

diff --git a/src/mesa/drivers/dri/i965/intel_screen.c b/src/mesa/drivers/dri/i965/intel_screen.c
index 21786eb..4acc1688 100644
--- a/src/mesa/drivers/dri/i965/intel_screen.c
+++ b/src/mesa/drivers/dri/i965/intel_screen.c
@@ -521,7 +521,7 @@ intel_create_image(__DRIscreen *dri_screen,
    int cpp;
    unsigned long pitch;
 
-   tiling = I915_TILING_X;
+   tiling = screen->devinfo.gen >= 9 ? I915_TILING_Y : I915_TILING_X;
    if (use & __DRI_IMAGE_USE_CURSOR) {
       if (width != 64 || height != 64)
 	 return NULL;
@@ -1180,7 +1180,7 @@ intel_detect_swizzling(struct intel_screen *screen)
    drm_intel_bo *buffer;
    unsigned long flags = 0;
    unsigned long aligned_pitch;
-   uint32_t tiling = I915_TILING_X;
+   uint32_t tiling = screen->devinfo.gen >= 9 ? I915_TILING_Y : I915_TILING_X;
    uint32_t swizzle_mode = 0;
 
    buffer = drm_intel_bo_alloc_tiled(screen->bufmgr, "swizzle test",
@@ -1794,7 +1794,7 @@ intelAllocateBuffer(__DRIscreen *dri_screen,
       return NULL;
 
    /* The front and back buffers are color buffers, which are X tiled. */
-   uint32_t tiling = I915_TILING_X;
+   uint32_t tiling = screen->devinfo.gen >= 9 ? I915_TILING_Y : I915_TILING_X;
    unsigned long pitch;
    int cpp = format / 8;
    intelBuffer->bo = drm_intel_bo_alloc_tiled(screen->bufmgr,
-- 
cgit v0.10.2

